client/source/js/modules/analysis/optimization-ctrl.js

```diff
@@ -275,10 +277,129 @@ define(['./module', 'angular', 'd3'], function (module, angular, d3) {
       }
     };
 
+    function validateYears(){
+      return $scope.params.objectives!==undefined &&
+             $scope.params.objectives.year!==undefined &&
+             $scope.params.objectives.year.start!==undefined &&
+             $scope.params.objectives.year.end !==undefined &&
+             ($scope.params.objectives.year.start < $scope.params.objectives.year.end);
+    }
+
+    /**
+     * If the string is undefined return empty, otherwise just return the string
+     * @param str
+     * @returns {string}
+     */
+    function strOrEmpty(str){
+      return _(str).isUndefined() ? '' : str;
```

Comment:
Mixed methods of defining `no value` cases.
In the controller vast usage of `=== undefined` but here underscore is used. And overall this function is obsolete as it can be replaced with: `str = str || '';`